### ISSUE TITLE
TypeError: comments.map is not a function (Unhandled Runtime Error)

### ISSUE THREAD

**Issue by upenr** [vercel/next.js#31504]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

14.16.1

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

local

### Describe the Bug

Followed the instructions here: https://github.com/vercel/next.js/tree/canary/examples/blog-with-comment

Ran it. On one of the posts under /posts (for example, posts/prancing-pony), wait a few seconds after the post loads. The following error appears, 

Unhandled Runtime Error
TypeError: comments.map is not a function

Source
.next\static\chunks\main.js (17356:0) @ updateFunctionComponent

  17354 |     ReactCurrentOwner$1.current = workInProgress;
  17355 |     setIsRendering(true);
> 17356 |     nextChildren = renderWithHooks(current, workInProgress, Component, nextProps, context, renderLanes);
  17357 | 
  17358 |     if ( workInProgress.mode & StrictMode) {
  17359 |       disableLogs();

### Expected Behavior

Caught exception? 

### To Reproduce

Followed the instructions here: https://github.com/vercel/next.js/tree/canary/examples/blog-with-comment

Ran it. On one of the posts under /posts (for example, posts/prancing-pony), wait a few seconds after the post loads. The aforementioned error appears.

**Comment by shavidze** [comment]
@upenr are you sure correctly set up the project, for example did you go to the Upstash Console and create a new database?
Or did you copy .env.local.example to .env.local
cp .env.local.example .env.local?
I almost believe the bug was caused by these things

**Comment by upenr** [comment]
Yes, @shavidze. Upstash database is created, and .env.local has 4 variables. 
Copied the UPSTASH_REDIS_REST_URL var from the Upstash Console into .env.local as REDIS_URL. NEXT_PUBLIC_AUTH0_CLIENT_ID, and NEXT_PUBLIC_AUTH0_DOMAIN were from Auth0. Signed in to Auth0 and Upstash with GitHub, but used the email associated with them for the NEXT_PUBLIC_AUTH0_ADMIN_EMAIL var.

**Comment by shavidze** [comment]
@upenr hmmm quite weird, Did you run yarn build before starting the application? It seems that you have done something wrong with Redis, that's why the API call returns a bad request.

**Comment by Franetse7** [comment]
Salut mon cher juste pour vous dire merci pour les correction

**Comment by upenr** [comment]
@shavidze It's being built and run locally with `npm run dev`.

**Comment by shavidze** [comment]
> @shavidze It's being built and run locally with `npm run dev`.

Could you share some more details, I tested it and it works fine for me.
![image](https://user-images.githubusercontent.com/23502763/142658518-84bf04f8-14a5-4f77-af24-e2e4d3c770a1.png)

**Comment by upenr** [comment]
Here's the behavior I am seeing @shavidze: https://imgur.com/a/0FpapJE

**Comment by shavidze** [comment]
I know that and the **GET comment API**  returns 400 error code itself.

I reproduced your case when I'd changed the Redis URL to the bad one. 

![image](https://user-images.githubusercontent.com/23502763/142662474-6dd90ef5-2ef5-494d-b960-f08a51e53c61.png)

**Comment by upenr** [comment]
Thanks. Isn't the REDIS_URL the endpoint shown in the Upstash Console? Or am I missing sth.? 

![image](https://user-images.githubusercontent.com/29108655/142666048-a5fa66ce-66fe-48df-96d6-86c9563bf814.png)

**Comment by upenr** [comment]
@shavidze Thanks. Isn't the REDIS_URL the endpoint shown in the Upstash Console? Or am I missing sth.?
![image](https://user-images.githubusercontent.com/29108655/142666408-7b4a9622-8c64-4167-96f5-b8f54173a022.png)

**Comment by upenr** [comment]
Ah, so I had to set the REDIS_URL to something like this - redis://:password@endpoint:port. This can be added to the README.

**Comment by upenr** [comment]
Ah, so I had to set the REDIS_URL to something like this - redis://:password@endpoint:port. This can be added to the README.

**Comment by upenr** [comment]
@shavidze Even after setting it to REDIS_URL=redis://:password@endpoint:port, I have the same issue. What's the right format of the REDIS_URL var? Thanks.

**Comment by shavidze** [comment]
![Screenshot_7](https://user-images.githubusercontent.com/23502763/142670200-614ac90d-6dc0-4d25-aceb-00b5fd15bb48.png)
Just copy this and paste it from **redis://.....** 
And before npm run dev just make sure that everything is built.( npm run build).

**Comment by upenr** [comment]
Yes @shavidze, that's exactly what I've done. Copy-pasted the URL from there, but I am still getting the same error. I have also run the `npm run build` before `npm run dev`. Strange. 
![image](https://user-images.githubusercontent.com/29108655/142672404-eb3c6232-7db8-4b89-b631-b06ae018700e.png)

**Comment by shavidze** [comment]
@upenr Can you provide your .env.local?

**Comment by upenr** [comment]
This is how it looks @shavidze;
REDIS_URL=redis://password@endpoint:port
NEXT_PUBLIC_AUTH0_CLIENT_ID=<copied-from-autho0-single-page-app>
NEXT_PUBLIC_AUTH0_DOMAIN=dev-xxxxxx-2.us.auth0.com
NEXT_PUBLIC_AUTH0_ADMIN_EMAIL=<email-id-that-shows-in-auth0>

**Comment by upenr** [comment]
@shavidze I think I've got it. The TLS (SSL) option was enabled on the database. I disabled it and it seems to work now. Is there an example with TLS (SSL) enabled on the database?

**Comment by shavidze** [comment]
Congrats at first.🎊
I don't really know 
**This is what I read at upstash**

_redis-cli supports TLS starting with version 6. If you are using version 5 or earlier, you should use Stunnel to establish a secure connection to the TLS enabled database_

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
